ksqldb-parser/src/test/java/io/confluent/ksql/parser/AstBuilderTest.java

```diff
@@ -1142,4 +1150,240 @@ public void shouldThrowOnNonIntegerId() {
+  }
+
+  @Test
+  public void shouldGetCorrectLocationsSimpleCsas() {
+    // Given:
+    final String query = "CREATE STREAM X AS SELECT * FROM TEST1;";
+    final SingleStatementContext stmt = givenQuery(query);
+
+    // When:
+    final CreateStreamAsSelect result = (CreateStreamAsSelect) builder.buildStatement(stmt);
+
+    // Then:
+    final NodeLocation loc = result.getLocation().get();
+    assertThat(loc.getStartLineNumber(), is(1));
+    assertThat(loc.getStartColumnNumber(), is(1));
+    assertThat(loc.getStopLine(), is(OptionalInt.of(1)));
+    assertThat(loc.getStopColumnNumber(), is(OptionalInt.of(query.length() - 1)));
+    assertThat(loc.getLength(), is(OptionalInt.of(query.length() - 1)));
+  }
+
+  @Test
+  public void shouldGetCorrectLocationsSimpleCsasMultiline() {
+    // Given:
+    final String query =
+        "CREATE STREAM X AS\n" +
+        "  SELECT * FROM TEST1;";
+    final SingleStatementContext stmt = givenQuery(query);
+
+    // When:
+    final CreateStreamAsSelect result = (CreateStreamAsSelect) builder.buildStatement(stmt);
+
+    // Then:
+    final NodeLocation loc = result.getLocation().get();
+    assertThat(loc.getStartLineNumber(), is(1));
+    assertThat(loc.getStartColumnNumber(), is(1));
+    assertThat(loc.getStopLine(), is(OptionalInt.of(2)));
+    assertThat(loc.getStopColumnNumber(), is(OptionalInt.of("  SELECT * FROM TEST1".length())));
+    assertThat(loc.getLength(), is(OptionalInt.of(query.length() - 1)));
+  }
+
+  @Test
+  public void shouldGetCorrectLocationsSimpleCt() {
+    // Given:
+    final String query = "CREATE TABLE X WITH (kafka_topic='X');";
+    final SingleStatementContext stmt = givenQuery(query);
+
+    // When:
+    final CreateTable result = (CreateTable) builder.buildStatement(stmt);
+
+    // Then:
+    final NodeLocation loc = result.getLocation().get();
+    assertThat(loc.getStartLineNumber(), is(1));
+    assertThat(loc.getStartColumnNumber(), is(1));
+    assertThat(loc.getStopLine(), is(OptionalInt.of(1)));
+    assertThat(loc.getStopColumnNumber(), is(OptionalInt.of(query.length() - 1)));
+    assertThat(loc.getLength(), is(OptionalInt.of(query.length() - 1)));
+  }
+
+  @Test
+  public void shouldGetCorrectLocationsSimpleCtas() {
+    // Given:
+    final String query = "CREATE TABLE X AS SELECT COUNT(1) FROM TEST1 GROUP BY ROWKEY;";
+    final SingleStatementContext stmt = givenQuery(query);
+
+    // When:
+    final CreateTableAsSelect result = (CreateTableAsSelect) builder.buildStatement(stmt);
+
+    // Then:
+    final NodeLocation loc = result.getLocation().get();
+    assertThat(loc.getStartLineNumber(), is(1));
+    assertThat(loc.getStartColumnNumber(), is(1));
+    assertThat(loc.getStopLine(), is(OptionalInt.of(1)));
+    assertThat(loc.getStopColumnNumber(), is(OptionalInt.of(query.length() - 1)));
+    assertThat(loc.getLength(), is(OptionalInt.of(query.length() - 1)));
+  }
+
+  @Test
+  public void shouldGetCorrectLocationsSimpleCtasMultiLine() {
```

Comment:
ditto (though perhaps keep this one instead of the multi-line CSAS because this has more lines so it's a bit more interesting)